crm_lead_firstname/models/crm_lead.py

```diff
@@ -0,0 +1,44 @@
+# -*- coding: utf-8 -*-
+# © 2016 Antiun Ingeniería S.L. - Jairo Llopis
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+
+from collections import OrderedDict
+from openerp import api, fields, models
+
+
+class CrmLead(models.Model):
+    _inherit = "crm.lead"
+
+    contact_name = fields.Char("First name")
+    contact_lastname = fields.Char("Last name")
+
+    @api.model
+    def _lead_create_contact(self, lead, name, is_company, parent_id=False):
+        """Ensure first and last names of contact match those in lead."""
+        partner_id = super(CrmLead, self)._lead_create_contact(
+            lead, name, is_company, parent_id)
+        if not is_company and partner_id:
+            partner = self.env["res.partner"].browse(partner_id)
+            partner.update(
+                OrderedDict(
```

Comment:
You should write the same commit text as comment for clarifying it to the next code reader